src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -195,8 +198,16 @@ object ResourceMatcher extends StrictLogging {
       badConstraints.isEmpty
     }
 
+    val checkAvailability: Boolean = {
+      if (conf.availableFeatures.contains(Features.MAINTENANCE_MODE))
+        Availability.offerAvailable(offer, conf.drainingTime)
```

Comment:
We need logging why this offer was declined.

Reply:
And we need to add new statistic for declined offers due to `MaintenanceMode` (the `v2/queue`)